GitWorkflow/GitWorkflow/Player.cpp

```diff
@@ -0,0 +1,77 @@
+#include "Player.hpp"
+
+Player::Player(const sf::Vector2f& position, const sf::Vector2f& size, const sf::Color& color)
+	: Drawable(), m_model(size)
+{
+	m_model.setFillColor(color);
+	m_model.setPosition(position);
+}
+
+void Player::move(const sf::Vector2f& offset)
+{
+	m_model.move(offset);
+}
+
+void Player::setPosition(const sf::Vector2f& position)
+{
+	m_model.setPosition(position);
+}
+
+const sf::Vector2f& Player::getPosition() const
+{
+	return m_model.getPosition();
+}
+
+void Player::setVelocity(const sf::Vector2f& velocity)
+{
+	m_velocity = velocity;
+}
+
+const sf::Vector2f& Player::getVelocity() const
+{
+	return m_velocity;
+}
+
+const sf::FloatRect Player::getGlobalBounds() const
+{
+	return m_model.getGlobalBounds();
+}
+
+void Player::collide(const Platform& platform)
+{
+	auto intersection = util::intersection(getGlobalBounds(), platform.getGlobalBounds());
+	bool verticalCollision = abs(intersection.width) > abs(intersection.height);
+
+	move(verticalCollision ? sf::Vector2f(0.0f, intersection.height) : sf::Vector2f(intersection.width, 0.0f));
+
+	if (verticalCollision)
+	{
+		if(m_velocity.y > 0.0f) m_hasJumped = false;
+		setVelocity({ m_velocity.x, 0.0f });
+	}
+}
+
+void Player::update(const float dt)
+{
+	if (m_velocity.y == 0.0f)
```

Comment:
As discussed, the x-movement should not be restricted to only on the ground.
In the future, we could add a lesser x-movement change when in air than when on the ground, but some control should exist in the air.

Reply:
Somewhat fixed.